src/main/scala/internal/AnalysisUtils.scala

```diff
@@ -0,0 +1,118 @@
+package internal
+
+object AnalysisUtils {
+
+  /**
+   * Given a graph and a source/sink/node, output the set of all sources/sinks/nodes
+   * that directly depend on the source/sink/node.
+   *
+   * i.e. All ids with a direct dependency on this source/sink/node.
+   *
+   * @param graph The graph to use
+   * @param id The node/sink/source in the graph
+   * @return The set of all children of that id
+   */
+  def getChildren(graph: Graph, id: GraphId): Set[GraphId] = {
+    id match {
+      case id: NodeOrSourceId => {
+        val childrenNodes = graph.dependencies.filter(_._2.contains(id)).keySet
```

Comment:
I don't see this becoming a problem in the short term but we might want to point out that this is `O(VE)` and that it could be done faster with explicit bookkeeping in the `Graph` class pointing in both directions.